spec/models/claim_spec.rb

```diff
@@ -468,4 +468,24 @@
       expect(Claim::FILTER_PARAMS.keys).to match_array(Claim.new.attribute_names.map(&:to_sym))
     end
   end
+
+  describe "payrollable" do
+    let(:payroll_run) { create(:payroll_run, claims_count: 1) }
+    let!(:submitted_claim) { create(:claim, :submitted) }
+    let!(:first_unpayrolled_claim) { create(:claim, :approved) }
+    let!(:second_unpayrolled_claim) { create(:claim, :approved) }
+
+    it "includes claims that do not belong to a payroll run" do
```

Comment:
It's written the way it is so that there's an `it` for each of the method's behaviours that I considered important. You're right, though - all of the desired behaviours can be exhibited simultaneously by a single set of test data. I just wonder how to write a single `it` that makes all of the desired behaviours clear, without a huge `it `description. I guess it could use comments instead, like

```ruby
it "returns the correct claims" do
  payrollable = described_class.payrollable

  # includes claims that do not belong to a payroll run
  expect(payrollable).to include(first_unpayrolled_claim)
  expect(payrollable).to include(second_unpayrolled_claim)

  # does not include claims that belong to a payroll run
  expect(payrollable).not_to include(payroll_run.claims.first)

  # only includes approved claims
  expect(payrollable).not_to include(submitted_claim)
end
```

What do you think? Also, I should probably introduce an extra commit to introduce this restructuring of the spec; it isn't directly related to any of the other changes made here.

Reply:
Or, did you literally want something like this?

```ruby
it "returns the correct claims" do
  expect(described_class.payrollable).to match_array([first_unpayrolled_claim, second_unpayrolled_claim])
end
```

Reply:
I reckon the latter is much easier to follow 👍

Reply:
Yep, the latter, but with a more helpful description for the spec. Say “it returns approved claims that are not associated with a payroll run”

Reply:
Sure thing, thanks.